lib/attachment.js

```diff
@@ -28,18 +28,24 @@ module.exports = function (schema, options) {
 
             var item = items[i];
 
+            // debug purposes
+            if ((schema.tree[item].type == Mongoose.SchemaTypes.Image)) {
+                // console.log("Schema.tree[item]:", schema.tree[item]);
+                // console.log("Instance[item]:", instance[item]);
+                // console.log("Instance.isModified(item):", instance.isModified(item));
+            }
+
             if ((schema.tree[item].type == Mongoose.SchemaTypes.Image) && instance[item] && (instance[item] != "") && instance.isModified(item)) {
                 if (instance[item].filename && instance[item].path) {
 
                     var extension;
                     try {
-                        extension = instance[item]["headers"]["content-type"].split("/")[1];
+                        extension = instance[key]["headers"]["content-type"].split("/")[1];
```

Comment:
Is using `key` as accessor valid here (not using `item`)?

Reply:
my fault, :/

Reply:
Hey, english2go.

Have not you guys already forked this repo?

Reply:
Hey Ugur abi :) 
yes, we already forked this repo :) but didn't let me push from forked repo:/ 
i will publish new patch for italy version

Reply:
Hey Uğur abi. 
I haven't experienced publishing from forked repo,
I thought it would be better to publish new patch from original repo, i could be wrong :/